src/protocol/taiko_alethia/ProverManager.sol

```diff
@@ -0,0 +1,357 @@
+    function payPublicationFee(address proposer, bool isDelayed) external payable {
+        require(msg.sender == inbox, "Only the Inbox contract can call this function");
+
+        // Accept additional deposit if sent
+        if (msg.value > 0) {
+            _deposit(proposer, msg.value);
+        }
+
+        uint256 periodId = currentPeriodId;
+
+        uint256 periodEnd = _periods[periodId].end;
+        if (periodEnd != 0 && block.timestamp > periodEnd) {
+            // Advance to the next period
+            currentPeriodId = ++periodId;
+            emit NewPeriod(periodId);
+        }
+
+        // Deduct fee from proposer's balance
+        balances[proposer] -= _periods[periodId].fee;
+    }
+
+    /// @inheritdoc IProverManager
+    /// @dev The offered fee has to be at least `minUndercutPercentage` lower than the current best price.
+    /// @dev The current best price may be the current prover's fee or the fee of the next bid, depending on whether the
+    /// period is active or not.
+    /// An active period is one that doesn't have an `end` timestamp yet.
+    function bid(uint256 offeredFee) external {
+        uint256 currentPeriod = currentPeriodId;
+        Period storage _currentPeriod = _periods[currentPeriod];
+        Period storage _nextPeriod = _periods[currentPeriod + 1];
+        uint256 requiredMaxFee;
+        if (_currentPeriod.end == 0) {
+            // If the period is still active the bid has to be lower
+            uint256 currentFee = _currentPeriod.fee;
+            requiredMaxFee = _calculatePercentage(currentFee, maxBidPercentage);
+            require(offeredFee <= requiredMaxFee, "Offered fee not low enough");
+
+            uint256 periodEnd = block.timestamp + successionDelay;
+            _finalizePeriod(_currentPeriod, periodEnd, periodEnd + provingWindow);
+        } else {
+            address _nextProverAddress = _nextPeriod.prover;
+            if (_nextProverAddress != address(0)) {
+                // If there's already a bid for the next period the bid has to be lower
+                uint256 nextFee = _nextPeriod.fee;
+                requiredMaxFee = _calculatePercentage(nextFee, maxBidPercentage);
+                require(offeredFee <= requiredMaxFee, "Offered fee not low enough");
+
+                // Refund the liveness bond to the losing bid
+                balances[_nextProverAddress] += _nextPeriod.stake;
+            }
+        }
+
+        // Record the next period info
+        uint256 _livenessBond = livenessBond;
+        _updatePeriod(_nextPeriod, msg.sender, offeredFee, _livenessBond);
+
+        emit ProverOffer(msg.sender, currentPeriod + 1, offeredFee, _livenessBond);
+    }
+
+    /// @inheritdoc IProverManager
+    /// @dev This can be called by anyone, and they get `evictorIncentivePercentage` of the liveness bond as an
+    /// incentive.
+    function evictProver(IPublicationFeed.PublicationHeader calldata publicationHeader) external {
```

Comment:
Looking at the eviction logic, it seems that when a prover is evicted (or slashed for inactivity), their staked liveness bond remains in the contract indefinitely, with no immediate way to reclaim or redistribute it. In `evictProver`, after paying out the evictor incentive, the remaining stake stays in `period.stake`.

If no one finalizes the period (i.e., no call to `finalizeClosedPeriod`), the leftover bond just sits locked. Is this the intended behavior? It effectively means the slashed prover’s funds are stuck unless a new prover completes the job. 

Should there be a fallback mechanism to handle abandoned stake in cases where no one follows up?